streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -3007,7 +3007,22 @@ abstract class ZStream[-R, +E, +O](val process: ZManaged[R, Nothing, ZIO[R, Opti
    */
   def zipAllWith[R1 <: R, E1 >: E, O2, O3](
     that: ZStream[R1, E1, O2]
-  )(left: O => O3, right: O2 => O3)(both: (O, O2) => O3): ZStream[R1, E1, O3] = {
+  )(left: O => O3, right: O2 => O3)(both: (O, O2) => O3): ZStream[R1, E1, O3] =
+    zipAllWithExec(that)(left, right)(both)(ExecutionStrategy.Parallel)
+
+  /**
+   * Zips this stream with another point-wise. The provided functions will be used to create elements
+   * for the composed stream.
+   *
+   * The functions `left` and `right` will be used if the streams have different lengths
+   * and one of the streams has ended before the other.
+   *
+   * The execution strategy `exec` will be used to determine whether to pull
+   * from the streams sequentially or in parallel.
+   */
+  def zipAllWithExec[R1 <: R, E1 >: E, O2, O3](
+    that: ZStream[R1, E1, O2]
+  )(left: O => O3, right: O2 => O3)(both: (O, O2) => O3)(exec: ExecutionStrategy): ZStream[R1, E1, O3] = {
```

Comment:
Should we place the strategy’s parameter list first so that the functions are still specified last?

Reply:
Good idea! Will change now.